plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -543,6 +544,7 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                               ("block_num", chain.head_block_num() + 1)
                               ("prod", chain.pending_block_producer())
                               ("txid", trx->id()));
+                     schedule_maybe_produce_block( true );
```

Comment:
To avoid a large trx triggering this and stopping block production when smaller trx can still fit, need some mechanism to say we are close enough to `max_block_*_usage` and stop trying trxs. Maybe a config option?